app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt

```diff
@@ -48,7 +48,9 @@ class ExternalPlayer(private val activity: MainActivity) : KoinComponent {
     }
 
     private fun notifyEvent(event: String, parameters: String = "") {
-        activity.loadUrl("javascript:window.ExtPlayer.notify$event($parameters)")
+        activity.runOnUiThread {
+            activity.loadUrl("javascript:window.ExtPlayer.notify$event($parameters)")
```

Comment:
We might want to change this to use JSON or something else because right now we can use an XSS attack from within the Kotlin code.

Given the event `(); alert('example'); noop(` it will fail

Reply:
@nielsvanvelzen the same already happens in the ExoPlayer component however. The question is whether we can ignore it, because the function is private and only called by us anyway, or if we should add some sanitization, like checking is every char is a letter.

Reply:
Better be safe then sorry

Reply:
Yep, the same can happen in https://github.com/jellyfin/jellyfin-android/blob/39accd32e0b6dd0b3830e62414e0f4008dff966e/app/src/main/java/org/jellyfin/mobile/MainActivity.kt#L119

```
Function
    loadUrl(String)
Found usages  (6 usages found)
    Function call  (6 usages found)
        app  (6 usages found)
            org.jellyfin.mobile  (1 usage found)
                MainActivity.kt  (1 usage found)
                    MainActivity  (1 usage found)
                        onCreate  (1 usage found)
                            119 loadUrl("javascript:$function")
            org.jellyfin.mobile.bridge  (3 usages found)
                Commands.kt  (1 usage found)
                    Commands  (1 usage found)
                        triggerInputManagerAction  (1 usage found)
                            13 loadUrl(buildInputManagerCommand("trigger('$action')"))
                ExternalPlayer.kt  (1 usage found)
                    ExternalPlayer  (1 usage found)
                        notifyEvent  (1 usage found)
                            54 activity.loadUrl("javascript:window.ExtPlayer.notify$event($parameters)")
                NativeInterface.kt  (1 usage found)
                    NativeInterface  (1 usage found)
                        execCast  (1 usage found)
                            160 activity.loadUrl("""javascript:window.NativeShell.castCallback("$action", $keep, $err, $result);""")
            org.jellyfin.mobile.webapp  (2 usages found)
                RemotePlayerService.kt  (2 usages found)
                    RemotePlayerService  (2 usages found)
                        ServiceBinder  (2 usages found)
                            436 webViewController?.loadUrl(Commands.buildInputManagerCommand("trigger('seek', $pos)"))
                            440 webViewController?.loadUrl(Commands.buildPlaybackManagerCommand("sendCommand({Name:'SetVolume', Arguments:{Volume:$value}})"))
```

for now I added some validations inside the function

Reply:
That would be something that I have to look into at a later time then..